Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift

```diff
@@ -21,20 +51,170 @@ public final class SwiftPackageManagerGraphGenerator: SwiftPackageManagerGraphGe
-            result[packageFolder.basename] = packageInfo
+            return (
+                name: name,
+                folder: packageFolder,
+                artifactsFolder: path.appending(component: "artifacts").appending(component: name),
+                info: packageInfo
+            )
         }
 
-        let thirdPartyDependencies = Dictionary(uniqueKeysWithValues: try packageInfos.map { name, packageInfo in
-            (name, try Self.mapToThirdPartyDependency(name: name, packageInfo: packageInfo))
+        let productToPackage: [String: String] = packageInfos.reduce(into: [:]) { result, packageInfo in
+            packageInfo.info.products.forEach { result[$0.name] = packageInfo.name }
+        }
+
+        let thirdPartyDependencies: [String: ThirdPartyDependency] = Dictionary(uniqueKeysWithValues: try packageInfos.map { packageInfo in
+            let dependency = try Self.mapToThirdPartyDependency(
+                name: packageInfo.name,
+                folder: packageInfo.folder,
+                info: packageInfo.info,
+                artifactsFolder: packageInfo.artifactsFolder,
+                productToPackage: productToPackage
+            )
+            return (dependency.name, dependency)
         })
 
         return DependenciesGraph(thirdPartyDependencies: thirdPartyDependencies)
     }
 
-    private static func mapToThirdPartyDependency(name: String, packageInfo _: PackageInfo) throws -> ThirdPartyDependency {
-        // TODO: map `PackageInfo` to actual `ThirdPartyDependency`
-        return .xcframework(name: name, path: .root, architectures: [])
+    private static func mapToThirdPartyDependency(
+        name: String,
+        folder: AbsolutePath,
+        info: PackageInfo,
+        artifactsFolder: AbsolutePath,
+        productToPackage: [String: String]
+    ) throws -> ThirdPartyDependency {
+        let products: [ThirdPartyDependency.Product] = info.products.compactMap { product in
+            guard let libraryType = product.type.libraryType else { return nil }
+
+            return .init(
+                name: product.name,
+                targets: product.targets,
+                libraryType: libraryType
+            )
+        }
+
+        let targets: [ThirdPartyDependency.Target] = try info.targets.compactMap { target in
+            switch target.type {
+            case .regular:
+                break
+            case .executable, .test, .system, .binary, .plugin:
+                logger.debug("Target \(target.name) of type \(target.type) ignored")
+                return nil
+            }
+
+            let path = folder.appending(RelativePath(target.path ?? "Sources/\(target.name)"))
+            let sources: [AbsolutePath]
+            if let customSources = target.sources {
+                sources = customSources.map { path.appending(RelativePath($0)) }
+            } else {
+                sources = [path]
```

Comment:
is it valid to have a single path as sources?

Reply:
yep, if you just specify the path (and not the `sources` property), then the sources will be found in that path (e.g. [Alamofire](https://github.com/Alamofire/Alamofire/blob/master/Package.swift))